production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala

```diff
@@ -0,0 +1,44 @@
+package dev.atedeg.mdm.productionplanning.dto
+
+import java.time.LocalDate
+
+import dev.atedeg.mdm.productionplanning.*
+import dev.atedeg.mdm.productionplanning.IncomingEvent.*
+import dev.atedeg.mdm.productionplanning.OutgoingEvent.*
+import dev.atedeg.mdm.productionplanning.dto.OrderIDDTO.given
+import dev.atedeg.mdm.productionplanning.dto.QuantityDTO.given
+import dev.atedeg.mdm.products.dto.ProductDTO
+import dev.atedeg.mdm.products.dto.ProductDTO.given
+import dev.atedeg.mdm.utils.serialization.DTO
+import dev.atedeg.mdm.utils.serialization.DTOGenerators.*
+import dev.atedeg.mdm.utils.serialization.DTOOps.*
+
+// incoming events
+final case class NewOrderReceivedDTO(order: OrderDTO)
+final case class OrderDTO(orderID: String, requiredBy: String, orderedProducts: List[OrderedProductDTO])
+final case class OrderedProductDTO(product: ProductDTO, quantity: Int)
+
+object NewOrderReceivedDTO:
+  given DTO[NewOrderReceived, NewOrderReceivedDTO] = interCaseClassDTO
+  private given DTO[Order, OrderDTO] = interCaseClassDTO
+  private given DTO[OrderedProduct, OrderedProductDTO] = interCaseClassDTO
+
+// outgoing events
+final case class ProductionPlanReadyDTO(productionPlan: ProductionPlanDTO)
+final case class ProductionPlanDTO(productsToProduce: List[ProductToProduceDTO])
```

Comment:
You do not need this intermediate passage for the dto, simply 
```scala 
final case class ProductionPlanReadyDTO(productionPlan: List[ProductToProduceDTO])
```
could suffice; @nicolasfara which one do you prefer?